docs/old-reference-guide/modules/messaging-concepts/pages/timeouts.adoc

```diff
@@ -0,0 +1,233 @@
+= Timeouts
+
+Axon Framework is built around the asynchronous processing of messages.
+In order to prevent a message from being processed indefinitely, Axon provides a mechanism to set a timeout on the invocation of a message handler, or on the processing of the `UnitOfWork` as a whole.
+
+== Overview
+
+Both timeouts work with a limit of the execution time (`timeoutMs`), and a time from which warnings will be logged (`warningThreshold`).
+When having a long-running handler or transaction, the following will happen:
+
+. After the `warningThreshold` has passed, a warning will be logged.
+. For each `warningInterval` that passes after the `warningThreshold`, another warning will be logged.
+. When the `timeoutMs` has passed, the handler or transaction will be interrupted.
+
+Both warnings and timeouts are logged at the `WARN` level.
+The message will contain:
+
+. The name of the handler with the message, or component of transaction
+. The time it has been processing so far
+. The time it has left before the timeout is reached
+. The stack trace of the handler or transaction, starting from where the timeout started
+
+For example, a warning message could look like this:
+
+[source,text]
+----
+2025-02-12T20:01:20.795Z  WARN 68040 --- [playground-correlation] [ axon-janitor-0] axon-janitor                             : Message [io.axoniq.playground.publisher.MyContinuousEvent] for handler [io.axoniq.playground.publisher.PersistentStreamEventProcessor] is taking long time to process. Current time: [5000ms]. Will be interrupted in [5000ms].
```

Comment:
```suggestion
2025-02-12T20:01:20.795Z  WARN 68040 --- [playground-correlation] [ axon-janitor-0] axon-janitor                             : Message [io.axoniq.playground.publisher.MyContinuousEvent] for handler [io.axoniq.playground.publisher.PersistentStreamEventProcessor] is taking a long time to process. Current time: [5000ms]. Will be interrupted in [5000ms].
```

Reply:
Well, wasn't a typo in the docs but a typo in all log statements. Corrected! Nice catch